tests/backup_helper.go

```diff
@@ -6933,3 +6933,74 @@ func ValidateCustomResourceRestores(ctx context1.Context, orgID string, resource
 	}
 	return nil
 }
+
+// func CheckBackupStatus return backup status and reason for a particular backup object.
+func CheckBackupStatus(backupName string, ctx context1.Context) (api.BackupInfo_StatusInfo_Status, string, error) {
+	backupDriver := Inst().Backup
+	backupUid, err := backupDriver.GetBackupUID(ctx, backupName, BackupOrgID)
+	if err != nil {
+		return 0, "", err
+	}
+	backupInspectRequest := &api.BackupInspectRequest{
+		Name:  backupName,
+		Uid:   backupUid,
+		OrgId: BackupOrgID,
+	}
+	resp, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+	if err != nil {
+		return 0, "", err
+	}
+	status := resp.GetBackup().GetStatus().Status
+	reason := resp.GetBackup().GetStatus().Reason
+	return status, reason, nil
+}
+
+// func InjectNetworkErrorInNodes inject error in application nondes by dropping packets or introdiucing delay in packet tramission.
+// nodes=> list of nodes where network injection should be done.
+// errorInjectionType => pass "delay" or "drop"
+// operationType => add/change/delete
+// dropPercentage => intger value from 1 to 100
+// delayInMilliseconds => 1 to 1000
+func InjectNetworkErrorInNodes(nodes []node.Node, errorInjectionType string, operationType string, dropPercentage int, delayInMilliseconds int) error {
```

Comment:
I feel this should be part of Inst().N

Reply:
we have similar utility InjectNetworkError in Inst().N , but this is more specific to backup

Reply:
I thought we could implement the "InjectNetworkErrorWithRebootFallback" method on Inst().N by reusing "InjectNetworkError". However, it seems that "RebootAndWait" is not implemented on Inst().N, suggesting that refactoring this might be beyond the scope of this PR.

If everything is in place, it would look something like this. I have not tested it.

```go
func (s *SSH) InjectNetworkErrorWithRebootFallback(nodes []node.Node, errorInjectionType string, operationType string, dropPercentage int, delayInMilliseconds int) error {
    for _, n := range nodes {
        err := s.InjectNetworkError([]node.Node{n}, errorInjectionType, operationType, dropPercentage, delayInMilliseconds)
        if err != nil {
            if strings.Contains(err.Error(), "no usable address found") {
                log.Infof("Node %s [is] unreachable. Rebooting the node to recover", n.Name)
                // recover the VM if it went unreachable by rebooting
                rebootErr := s.RebootNodeAndWait(n)
                if rebootErr != nil {
                    return fmt.Errorf("failed to reboot node %s: %v", n.Name, rebootErr)
                }
                log.Infof("Retrying network error injection on node [%s] after reboot", n.Name)
                err = s.InjectNetworkError([]node.Node{n}, errorInjectionType, operationType, dropPercentage, delayInMilliseconds)
                if err != nil {
                    return fmt.Errorf("failed to inject network error on node [%s] after reboot. Err: [%v]", n.Name, err)
                }
            } else {
                return err
            }
        }
    }
    return nil
}
```

Reply:
incorporated this changes , Thanks for suggestion @kphalgun-px

Reply:
Thank you.